hiclass/MultiLabelBinaryPolicy.py

```diff
@@ -0,0 +1,380 @@
+        # self.y = y.reshape(y.shape[0], -1)
+        self.sample_weight = sample_weight
+
+    def positive_examples(self, node) -> np.ndarray:
+        """
+        Gather all positive examples corresponding to the given node.
+
+        Parameters
+        ----------
+        node
+            Node for which the positive examples should be searched.
+
+        Returns
+        -------
+        positive_examples : np.ndarray
+            A mask for which examples are included (True) and which are not.
+        """
+        raise NotImplementedError
+
+    def negative_examples(self, node) -> np.ndarray:
+        """
+        Gather all negative examples corresponding to the given node.
+
+        Parameters
+        ----------
+        node
+            Node for which the negative examples should be searched.
+
+        Returns
+        -------
+        negative_samples : np.ndarray
+            A mask for which examples are included (True) and which are not.
+        """
+        raise NotImplementedError
+
+    def _get_descendants(self, node, inclusive: bool = True):
+        """
+        Gather all descendants for a given node.
+
+        Parameters
+        ----------
+        node
+            Node for which the descendants should be obtained.
+        inclusive : bool, default=True
+            True if the given node should be included in the list of descendants.
+
+        Returns
+        -------
+        descendants : set
+            Set of descendants for a given node.
+        """
+        descendants = set()
+        if inclusive:
+            descendants.add(node)
+        for successor in nx.dfs_successors(self.digraph, node).values():
+            descendants.update(successor)
+        return descendants
+
+    def _get_siblings(self, node):
+        """
+        Gather all siblings for a given node.
+
+        Parameters
+        ----------
+        node
+            Node for which the siblings should be obtained.
+
+        Returns
+        -------
+        siblings : set
+            Set of siblings for a given node.
+        """
+        parents = self.digraph.predecessors(node)
+        parents = list(parents)
+        siblings = set()
+        for parent in parents:
+            siblings.update(self.digraph.successors(parent))
+        siblings.discard(node)
+        return siblings
+
+    def get_binary_examples(self, node) -> tuple:
+        """
+        Gather all positive and negative examples for a given node.
+
+        Parameters
+        ----------
+        node
+            Node for which the positive and negative examples should be searched.
+
+        Returns
+        -------
+        X : np.ndarray
+            The subset with positive and negative features.
+        y : np.ndarray
+            The subset with positive and negative labels.
+        """
+        positive_examples = self.positive_examples(node)
+        negative_examples = self.negative_examples(node)
```

Comment:
```suggestion
        negative_examples = np.logical_and(
            np.logical_not(positive_examples),
            self.negative_examples(node)
        )
```
I think we should not have an example as both positive and negative. Perhaps here we could just update the code using logical operations, more or less like this. What are your thoughts?